version: '2.3'
services:
  app:
    image: ghcr.io/linuxserver/plex:1.40.5@sha256:5604745534016668a7a7debf2ad06eb95d06e0e27c99e8a10daf05c0a26f9559
    network_mode: host ###Leave this as it may detect your LAN traffic as WAN, forcing encoding. It may also run into issues if your internert is down
    #ports:
    #  - ${PLEXPORT:-32400}:32400
    #  - 1900:1900/udp
    #  - 3005:3005
    #  - 5353:5353/udp
    #  - 8324:8324
    #  - 32410:32410/udp
    #  - 32412:32412/udp
    #  - 32413:32413/udp
    #  - 32414:32414/udp
    #  - 32469:32469
    devices:
      - /dev/dri:/dev/dri
      #- /dev/dvb:/dev/dvb
      #- /dev/bus/usb/002/002:/dev/bus/usb/002/002
    environment:
      #- PUID=1000
      #- PGID=1000
      - VERSION=docker
      - PLEX_CLAIM=$PLEX_CLAIM
      - NVIDIA_VISIBLE_DEVICES=ALL
    runtime: nvidia
    volumes:
      - app:/config
      - media:/nfs/Media
      - /dev/shm:/dev/shm
    restart: unless-stopped
    healthcheck:
      test: curl --connect-timeout 15 --silent --show-error --fail http://localhost:32400/identity
      interval: 1m
      timeout: 30s
      retries: 3
      start_period: 1m
    labels:
      - "autoheal=true"
    logging:
      driver: "json-file"
      options:
        max-size: "25m"
        max-file: "3"
volumes:
  config:
  app:
  media:
    driver_opts:
      type: nfs 
      o: $STORAGE,nfsvers=4
      device: :$MEDIA
